packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -119,11 +119,78 @@ export function toRemovalInfo(maybe: Partial<IRemovalInfo> | undefined): IRemova
 	);
 }
 
+/**
+ * Tracks information about when and where this segment was moved to.
+ * @example - Suppose a merge tree had 3 TextSegments "X", "A", and "B", and
+ * received the operation `move({ start: 0, end: 1 }, { dest: 3 }, { seq: 30 })`
+ * (moving the "X" after the "A" and the "B").
+ * After processing this operation, it would have the segments
+ * `[<moved "X" tombstone>, "A", "B", "X"]`. The moved "X" tombstone segment
+ * would have the following IMoveInfo:
+ * `{ movedSeq: 30, moveDst: <reference to living "X" segment>}`
+ */
+export interface IMoveInfo {
```

Comment:
I would expect addition of this interface necessitates changes to `SnapshotV1.ts` at minimum, potentially also snapshotlegacy.ts (i'd guess not though since it serializes at MSN). either way, we should have some test verification of changes to serialized format due to obliterate with some behavioral verification of eventual consistency. things like this test:

```typescript
		it("can insert segments relative to removed segment loaded from snapshot", async () => {
			str.append("0x", /* increaseMsn: */ false);
			str.append("2", /* increaseMsn: */ false);
			str.removeRange(1, 2, /* increaseMsn: */ false);

			// Note that calling str.expect() switches the underlying client to the one loaded from the snapshot.
			await str.expect("02");

			str.insert(1, "1", /* increaseMsn: */ false);
			str.append("3", /* increaseMsn: */ false);
			await str.expect("0123");
		});
```

Reply:
changes in this realm and added tests look good for initial implementation. I filed AB#4009 as some light testing followup here.